### PR TITLE
fix http://www.github.com/wso2/product-apim/issues/7495

### PR DESCRIPTION
**Pull request by tharindu1st**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1739,7 +1739,7 @@
         <commons-logging.version>1.1.1</commons-logging.version>
         <commons-io.version>2.4.0.wso2v1</commons-io.version>
         <commons-httpclient.version>3.1</commons-httpclient.version>
-        <google.code.gson.version>2.8.6</google.code.gson.version>
+        <google.code.gson.version>2.8.5</google.code.gson.version>
```
Comment (harsha89) [inline pom.xml]: any reason for the downgrade?

File: components/apimgt/org.wso2.carbon.apimgt.jms.listener/src/main/java/org/wso2/carbon/apimgt/jms/listener/utils/JMSMessageListener.java [hunk 2]
```diff
@@ -283,12 +286,15 @@ private synchronized void handleBlockingMessage(Map<String, Object> map) {
...
             }
-        } else if (APIConstants.BLOCKING_CONDITIONS_IP.equals(condition)) {
+        } else if (APIConstants.BLOCKING_CONDITIONS_IP.equals(condition) ||
+                APIConstants.BLOCK_CONDITION_IP_RANGE.equals(condition)) {
             if (ThrottleConstants.TRUE.equals(conditionState)) {
-                ServiceReferenceHolder.getInstance().getThrottleDataHolder().addIplockingCondition(conditionValue,
-                        conditionValue);
+                
+                ServiceReferenceHolder.getInstance().getThrottleDataHolder().addIplockingCondition(tenantDomain,
+                        conditionId, conditionValue,condition);
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.jms.listener/src/main/java/org/wso2/carbon/apimgt/jms/listener/utils/JMSMessageListener.java]: formatting issue

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java [hunk 3]
```diff
@@ -12387,7 +12393,7 @@ public String addBlockConditions(String conditionType, String conditionValue, St
             APIMgtDBUtil.closeAllConnections(insertPreparedStatement, connection, null);
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java]: should do the closing in finally block?

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/throttling/ThrottleDataHolder.java [hunk 4]
```diff
@@ -226,11 +257,45 @@ public void removeKeyTemplate(String name) {
...
+                        }
+                    } else {
+                        if (ipRange.isInvert()) {
+                            return true;
+                        }
+                    }
+                } else if (APIConstants.BLOCK_CONDITION_IP_RANGE.equals(ipRange.getType())) {
+                    BigInteger ipBigIntegerValue = APIUtil.ipToBigInteger(ip);
+
+                    if (((ipBigIntegerValue.compareTo(ipRange.getStartingIpBigIntValue()) > 0) &&
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/throttling/ThrottleDataHolder.java]: will this work for both IPV4 and IPV6?

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/dto/IPRange.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
+/*
```
Comment (harsha89) [inline components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/dto/IPRange.java]: Generally add some debug logs to identofy issues.

File: pom.xml (hunk 1 above)
Comment (tharindu1st) [reply pom.xml]: this was not used in the pack an it was at installed state

File: components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/throttling/ThrottleDataHolder.java (hunk 4 above)
Comment (tharindu1st) [reply components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/throttling/ThrottleDataHolder.java]: yes

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java (hunk 3 above)
Comment (tharindu1st) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java]: its already in finally block

**Pull request merged**